File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        s = set(nums)
        l = 0
        r = 0
        n = len(nums)
        length = 1
        max_len = 0
        while r < n:
            c = nums[r]
            
            while c + 1 in s:
                length += 1
                c += 1
            max_len = max(length, max_len)
            length = 1
            r += 1
        return max_len

        

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-3.rs
impl Solution {
    pub fn max_profit(prices: Vec<i32>) -> i32 {
        let mut mp = 0;
        let n = prices.len();

        for i in 0..n {
            for j in i..n {
                mp = mp.max(prices[j] - prices[i]);
            }
            println!("{:?}", mp);
        }
        return mp;
    }
}
